src/crimson/CMakeLists.txt

```diff
@@ -11,6 +11,27 @@ set_target_properties(crimson::cflags PROPERTIES
   INTERFACE_COMPILE_OPTIONS $<$<COMPILE_LANGUAGE:CXX>:-Wno-non-virtual-dtor>
   INTERFACE_LINK_LIBRARIES Seastar::seastar)
 
+# sort out which allocator to use
+if(ALLOCATOR STREQUAL "tcmalloc")
```

Comment:
we already have almost the same logic at https://github.com/ceph/ceph/blob/249a57d28ae5e5977869644cefdbc7ddba0c2b1f/src/CMakeLists.txt#L280-L298

and `crimson/CMakeLists.txt` is included at https://github.com/ceph/ceph/blob/249a57d28ae5e5977869644cefdbc7ddba0c2b1f/src/CMakeLists.txt#L378 after these `add_compile_options()` calls.

bluestore bits are compiled at https://github.com/ceph/ceph/blob/master/src/crimson/os/alienstore/CMakeLists.txt as part of `https://github.com/ceph/ceph/blob/master/src/crimson/os/CMakeLists.txt` which is in turn included by this very file.

so i think as long as we have `-DALLOCATOR=tcmalloc` or `-DALLOCATOR=tcmalloc_minimal`, crimson and alienstore are both compiled with TCMalloc already. the same applies to JeMalloc.

Reply:
FWIW,  #46103 is created to cleanup this part a little bit.

Reply:
@tchaikov Sure, I have no strong opinion on the build scripts.  Does #46103 replace this PR?

Reply:
it does not. it is merely a cleanup. while your change does introduce a functional change which reduces the memory footprint dramatically. i just don't understand how it achieves its goal.

Reply:
Hmm, this is strange. I can't find any major user of `ALLOC_LIBS` besides `rgw` while quick googling hasn't indicated this variable is anyhow special.

```
$ grep -r ALLOC_LIBS
CMakeLists.txt:  set(ALLOC_LIBS gperftools::tcmalloc)
CMakeLists.txt:  set(ALLOC_LIBS gperftools::tcmalloc_minimal)
CMakeLists.txt:  set(ALLOC_LIBS JeMalloc::JeMalloc)
rocksdb/thirdparty.inc:  set(JEMALLOC_LIBS debug ${JEMALLOC_LIB_DEBUG} optimized ${JEMALLOC_LIB_RELEASE})
rocksdb/thirdparty.inc:  set (THIRDPARTY_LIBS ${THIRDPARTY_LIBS} ${JEMALLOC_LIBS})
tools/crimson/CMakeLists.txt:target_link_libraries(perf-async-msgr ceph-common global ${ALLOC_LIBS})
tools/ceph-dencoder/CMakeLists.txt:  ${ALLOC_LIBS})
tools/cephfs_mirror/CMakeLists.txt:  ${ALLOC_LIBS})
tools/rbd_mirror/CMakeLists.txt:  ${ALLOC_LIBS}
rgw/CMakeLists.txt:  ${ALLOC_LIBS})
rgw/CMakeLists.txt:  global ${ALLOC_LIBS})
test/crimson/CMakeLists.txt:target_link_libraries(test-seastar-messenger-peer ceph-common global ${ALLOC_LIBS})
```

How do we orchestrate the linking with `tcmaloc{,_minimal}.so`?

Reply:
FWIW, I just recompiled without this PR applied using:
```./do_cmake.sh -DCMAKE_BUILD_TYPE=RelWithDebInfo -DWITH_SEASTAR=ON```

and neither ceph-osd nor crimson-osd are linked against tcmalloc, but radosgw is.

Reply:
Yeah, it might be we aren't :-(. Compiling with `WITH_SEASTAR=OFF` and `ALLOCATOR=tcmalloc` (but it will take awhile on my laptop).

Reply:
@rzarzynski  had a good hypothesis that it's perfglue that's pulling tcmalloc in.  I just recompiled and indeed, we don't need this at all. If we simply remove "AND NOT WITH_SEASTAR" from perfglue's CMakeLists.txt we will link tcmalloc in both crimson and classic OSD when "WITH_SEASTAR=ON" is used.